test/one/profiler/test/Runner.java

```diff
@@ -0,0 +1,278 @@
+            return false;
+        }
+
+        if (jdkVersion.length > 0 && !Arrays.asList(jdkVersion).contains(currentJdkVersion)) {
+            return false;
+        }
+
+        if (jvm.length > 0 && !Arrays.asList(jvm).contains(currentJvm)) {
+            return false;
+        }
+
+        if (arch.length > 0 && !Arrays.asList(arch).contains(currentArch)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static void run(Method m, Boolean retainLog) throws Exception{
+        for (Test test : m.getAnnotationsByType(Test.class)) {
+            if (!enabled(test)) {
+                log.log(Level.FINE, "Skipped " + m.getDeclaringClass().getName() + '.' + m.getName());
+                continue;
+            }
+
+            log.log(Level.INFO, "Running " + m.getDeclaringClass().getName() + '.' + m.getName() + "...");
+            final TestProcess p;
+            
+            try {
+                p = new TestProcess(test, libExt, retainLog);
+            } catch (Throwable e) {
+                log.log(Level.WARNING, e.getMessage());
+                return;
+            }
+
+            try {
+                Object holder = (m.getModifiers() & Modifier.STATIC) == 0 ? m.getDeclaringClass().newInstance() : null;
+                m.invoke(holder, p);
+                log.info("OK");
+            } catch (Throwable e) {
+                System.out.println(e.getCause());
+                log.log(Level.WARNING, "Test failed " + e.getCause().getMessage(), e.getCause().getStackTrace());
+            } finally {
+                try {
+                    p.close();
+                } catch (Throwable e) {
+                    log.log(Level.SEVERE, "Testing framework test could not close properly. " + e.getMessage(), e.getStackTrace());
+                }
+            }
+        }
+    }
+
+    public static void run(Class<?> cls, Boolean retainLog) throws Exception{
+        for (Method m : cls.getMethods()) {
+            run(m, retainLog);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        String propertyName = "java.util.logging.ConsoleHandler.formatter";
+        String propertyValue = System.getProperty(propertyName);
+
+        if (propertyValue == null) {
+            System.setProperty("java.util.logging.ConsoleHandler.formatter", "java.util.logging.SimpleFormatter");
+        }
+        propertyName = "java.util.logging.SimpleFormatter.format";
+        propertyValue = System.getProperty(propertyName);
+        if (propertyValue == null) {
+            System.setProperty("java.util.logging.SimpleFormatter.format", "%4$s: %5$s%6$s%n");
+        }
+
+        if (args.length == 0) {
+            System.out.println("Usage: java " + Runner.class.getName() + " TestName ...");
+            System.exit(1);
+        }
+
+        Logger rootLogger = Logger.getLogger("");
+        String logLevel =  args[0];
```

Comment:
I don't want making log level a required argument. It can be a system property instead (set with `-D` command line option).
Probably, we don't need fine-grained configuration for the logging level at all: this can be either default (INFO) or verbose (ALL).

Reply:
It is not required from user perspective, the default is info. Or do you mean that you don't want it in the method signature?

Reply:
I mean, not the required argument of `Runner` class. E.g., in case I want to debug some test in IDE without executing `make`.